improvement(treewide): Remove pylint

pehala opened the pull request:
Pylint is no longer used, I think we can remove all the pylint disables. Ruff has less checks than pylint, if needed we can add more check Ruff to compensate (followup PR).

Warning: This might negatively affect backports, if we are backporting any line that was disabled, but I think that is small risk

### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [x] None needed

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

fruch commented:
It's only v15 that has pylint...

So I think we are good to go

fruch reviewed (dismissed):
LGTM

fruch commented:
If we won't backport it, it's gonna mess up any backport to branches

so I suggest backporting it to all branches

soyacz commented:
> If we won't backport it, it's gonna mess up any backport to branches
> 
> so I suggest backporting it to all branches

did we backported ruff to all supported branches (e.g. perf-v15?)

pehala commented:
> did we backported ruff to all supported branches (e.g. perf-v15?)

Based on what @fruch said, all except for this branch should support it.



> so I suggest backporting it to all branches

I added some, but I am not sure if I added correct ones, please correct them if you think they are wrong

pehala commented:
tests failed on jenkins issue

vponomaryov commented:
Better to make such huge updates that not really dependent on each other in separate PRs.
As of now it is not backportable, it is easy to get lost in the list of changes.

I would like to see the changes be split into batches, for example - by 10 modules.

pehala commented:
> Better to make such huge updates that not really dependent on each other in separate PRs. As of now it is not backportable, it is easy to get lost in the list of changes.
> 
> I would like to see the changes be split into batches, for example - by 10 modules.

I do not see logic in splitint by certain number of modules, it will generate too many needless PR (i.e. even more spam), the change is consise and even though it is big all changes relate to only this. I am fine with manually fixing the backport branches, it will be manual but not hard, I do not think it will take me more than 15 mins to fix a backport PR.

vponomaryov commented:
> > Better to make such huge updates that not really dependent on each other in separate PRs. As of now it is not backportable, it is easy to get lost in the list of changes.
> > I would like to see the changes be split into batches, for example - by 10 modules.
> 
> I do not see logic in splitint by certain number of modules, it will generate too many needless PR (i.e. even more spam), the change is consise and even though it is big all changes relate to only this. I am fine with manually fixing the backport branches, it will be manual but not hard, I do not think it will take me more than 15 mins to fix a backport PR.

5 hours have passed since your latest push and it already has merge conflict.
So, it is first reason. Changing half of the code base in a single PR must have serious reasons which are absent here.
We can remove pylint comments even 1 by 1. We are not even dependent on it's removal.

Another reason - possibility to get human reviews.
Who is going to review 2500 lines of code changes for stuff which get merge conflicts that soon?
And "just believe me it is ok and I didn't do any mistake removing some unexpected stuff" is not argument.

Doing smaller batches we make less problems.

pehala commented:
> > > Better to make such huge updates that not really dependent on each other in separate PRs. As of now it is not backportable, it is easy to get lost in the list of changes.
> > > I would like to see the changes be split into batches, for example - by 10 modules.
> > 
> > 
> > I do not see logic in splitint by certain number of modules, it will generate too many needless PR (i.e. even more spam), the change is consise and even though it is big all changes relate to only this. I am fine with manually fixing the backport branches, it will be manual but not hard, I do not think it will take me more than 15 mins to fix a backport PR.
> 
> 5 hours have passed since your latest push and it already has merge conflict. So, it is first reason. Changing half of the code base in a single PR must have serious reasons which are absent here. We can remove pylint comments even 1 by 1. We are not even dependent on it's removal.

There is no way of preventing merge conflicts, even if we split we will have merge conflicts, only in a fewer PR.
Splitting it into batches also creates PR spam, 10 batches creates 10x PR (around 60 for that matter), and it wont prevent merge conflicts either, I will just have to care fixing 5 PR instead of 1. This PR should not exist, because we never should have left the pylint disables in here for a year after removing the actual pylint.

Nothing is dependening on this PR and this PR is harmless as it lacks functional changes.

> 
> Another reason - possibility to get human reviews. Who is going to review 2500 lines of code changes for stuff which get merge conflicts that soon? And "just believe me it is ok and I didn't do any mistake removing some unexpected stuff" is not argument.

I doubt that review of this PR will take fewer than 10 minutes. It is pretty clear from the git diff showned in github that it indeed does not modify any code and thus there is no need to investigate impacts of each individual line. It is possible that I chnages something which is not that just removal of pylint disables, but even that would be easy to find as ANY change that is not easy to read from the diff is a mistake regardless of if it impacts anything and should be removed.

> 
> Doing smaller batches we make less problems.

Doing smaller batches will require (number of batches) times more attention and will take (number of batches) more time to complete with no clear benefit.

In my opinion, this PR can be approved and merge withing 20 mins, only thing that needs to be done is to synchronize (which would be my problem) 2 reviewers and one of them with merge rights and we can get it easily and noone has to care about it again, it might take a week or two before we synchronize but when we do I believe it is gonna be painless. Consequence of this is gonna be a one or two PR that will have conflcits from this but I do not see that as a problem (That would probably still happen even with smaller PRs). It is an equivalent of ripping the Band-Aid quickly vs ripping slowly, which would take weeks

If you really feel that strongly about this, give a final word and I will split it up to any desired number of batches

pehala commented:
v2:
  * Extracted some changed into separate PRs:
    * pylintrc: https://github.com/scylladb/scylla-cluster-tests/pull/10751
    * cluster*: https://github.com/scylladb/scylla-cluster-tests/pull/10753
    * nemesis: https://github.com/scylladb/scylla-cluster-tests/pull/10752
    * Everything else is still in this PR

@vponomaryov are you ok with this now?

pehala commented:
Rebased, could we get it merged?
